pystorcli2/controller/__init__.py

```diff
@@ -528,16 +607,28 @@ def __init__(self, binary='storcli64'):
         self._binary = binary
         self._storcli = StorCLI(binary)
 
+    @ property
+    def show(self) -> List[dict]:
+        """Static show output to allow getting info of static attributes
+        without the need to touch cached responses
+        Return:
+            list: list of dicts of controllers data from show command
+        """
+        if not getattr(self, '_show', None):
+            out = self._storcli.run(['show'], allow_error_codes=[
+                StorcliErrorCode.INCOMPLETE_FOREIGN_CONFIGURATION])
+            response = common.response_data(out)
+            if "Number of Controllers" in response and response["Number of Controllers"] == 0:
+                self._show = []
+            else:
+                self._show = common.response_data_subkey(
+                    out, ['System Overview', 'IT System Overview'])
+        return self._show
```

Comment:
What is exactly the point of this?
The cache is used to avoid duplicated calls, and this seems to be a custom cache over a "global" cache.
Don't see the point but another issue-maker to have multiple versions of the same "show".

Have you found an issue with the cache? Maybe that's what we must address instead of this

Reply:
I saw cache on pysmart... and I think it really improves performance without much coding effort.

In the other hand, I see your point. There are scenarios where singleton+global configs/cache can be harder than initially expected.
Maybe a cache-per-property may be much better (probably using custom decorator? and/or some external cache-lib decorator?)
I have to think about this

Reply:
I should check that !